src/Revamp.rei

```diff
@@ -31,7 +28,7 @@ module Compiled: {
   let count: (t, string) => int;
   let replace: (t, string => string, string) => string;
   let replaceByString: (t, string, string) => string; /* TODO: better name? */
-  let split: (t, string) => array(string); /* TODO: should be list string? */
+  let split: (t, string) => array(option(string)); /* TODO: should be list string? */
```

Comment:
I think the API here would be better if the implementation was changed instead to filter out the `None`s. That would break with JS behaviour of course, but I think it's very rare that this would be useful. There's an example of the behaviour here: https://github.com/rescript-lang/rescript-compiler/issues/3120#issuecomment-433637078

Reply:
I appreciate your allowing me to take a shot at this. The most recent commit is my attempt to update the api per your guidance. I am not at all sure that this is what you had in mind. It seems to work and the tests are passing.

Reply:
See https://github.com/glennsl/bs-revamp/pull/1#discussion_r594552674